hbase/src/main/java/com/yahoo/ycsb/db/HBaseClient10.java

```diff
@@ -246,17 +266,22 @@ public int read(String table, String key, Set<String> fields, HashMap<String,Byt
             //do nothing for now...need to understand HBase concurrency model better
             return ServerError;
         }
-
-        if (r.isEmpty()) {
+        if (multiGetBatchSize == 0) {
+          if (r[0].isEmpty()) {
             return NoMatchingRecord;
+          }
         }
-        for (Cell c : r.listCells()) {
-            result.put(Bytes.toString(CellUtil.cloneQualifier(c)),
-                    new ByteArrayByteIterator(CellUtil.cloneValue(c)));
-            if (_debug) {
-                System.out.println("Result for field: "+Bytes.toString(CellUtil.cloneQualifier(c))+
-                        " is: "+Bytes.toString(CellUtil.cloneValue(c)));
+        for (Result res : r) {
+          if(!res.isEmpty()) {
```

Comment:
sorry, I'm still not seeing how this doesn't fail in the batch case.

take, for example, multiGetBatchSize <= 50, multiGets.size <= 0 (for the first time read is called)

1. we init r to a length 1 array of Result objects (each of those objects are null initially
2. the multiGets block activates (because multiGetBatchSize > 0)
3. we add the Get to the multiGets list
4. multiGets.size is now 1, which is less than 50, so we don't do the multiget call
5. multiGetBatchSize is still >0, so we skip the degenerate case check
6. we get to the for loop, which will iterator over the Result list (still size 1)
7. res will be the first reference, which is null
8. res.isEmpty explodes

I'd really like a test for this change, but our total lack of tests for hte HBase binding makes that a really steep request.

I haven't looked at #392 in some time; @cmccoy is it basically ready to go? I could review it quickly today and then submit a basic set of tests for the hbase-1.0+ binding.